### ISSUE TITLE
AmadeusClosestAirport tool should accept any LLM

### ISSUE THREAD

**Issue by mspronesti** [langchain-ai/langchain#15847]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.

### Description

`AmadeusClosestAirport` contains a "hardcoded" call to `ChatOpenAI` (see [here](https://github.com/langchain-ai/langchain/blob/a06db53c37344b5a9906fbf656173c3421109398/libs/community/langchain_community/tools/amadeus/closest_airport.py#L50)), while it would make sense to use the same `llm` passed to the chain/agent when initialized.

In addition, this implies that `AmadeusToolkit` implicitly depends on `openai`, which should not be the case.

Example (source code from the [docs](https://python.langchain.com/docs/integrations/toolkits/amadeus))
```py
from langchain_community.agent_toolkits.amadeus.toolkit import AmadeusToolkit
# Set environmental variables here
import os

os.environ["AMADEUS_CLIENT_ID"] = "CLIENT_ID"
os.environ["AMADEUS_CLIENT_SECRET"] = "CLIENT_SECRET"
os.environ["OPENAI_API_KEY"] = "API_KEY"
# os.environ["AMADEUS_HOSTNAME"] = "production" or "test"

toolkit = AmadeusToolkit() 
tools = toolkit.get_tools()

llm = OpenAI(temperature=0) # this can be any `BaseLLM`
agent = initialize_agent(
    tools=tools,
    llm=llm,
    verbose=False,
    agent=AgentType.STRUCTURED_CHAT_ZERO_SHOT_REACT_DESCRIPTION,
)

#  ==> agent calls `ChatOpenAI` regardless of `llm` <===
agent.run("What is the name of the airport in Cali, Colombia?")
```

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [X] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

**Comment by S-T-A-R-L-O-R-D** [comment]
I would like work on this, I am thinking of making a class and having an default argument for any basellm (default arg can be openAI) and if user want to use some other model then user can specify same while initialize class object.

**Comment by mspronesti** [comment]
Solved in #15879.
